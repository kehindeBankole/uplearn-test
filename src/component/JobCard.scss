@import "../scss/index.scss";
.job-box {
    background-color: $box;
    border-radius: 1em;
    width: 15rem;
    height: 15rem;
    padding: 1rem;
    color:white;
    .title {
      color: white;
      text-align: center;
    }
    .company-name{
        color: white;
    }
  }