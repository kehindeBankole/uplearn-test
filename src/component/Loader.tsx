import React from 'react';
import './Loader.scss'
function Loader() {
  return <div>
      <div className="loader-box" data-testid="loader-box"><p>Loading...</p></div>;
  </div>;
}

export default Loader;
