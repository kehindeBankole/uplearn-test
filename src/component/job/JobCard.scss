@import "../../scss/index.scss";

.job-box {
    background-color: $box;
    border-radius: 1em;
    width: 15rem;
    height: 15rem;
    padding: 1rem;
    color:white;
    cursor: pointer;
    .title {
      color: white;
      text-align: center;
    }
    .company-name{
        color: white;
    }
  }
  Link{
    text-decoration: none;
  }