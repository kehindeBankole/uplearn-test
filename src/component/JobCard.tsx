import "./JobCard.scss";
import { useNavigate } from "react-router-dom";

function JobCard({
  item,
}: {
  item: {
    slug: string;
    title: string;
    countries: { name: string; id: string }[];
    remotes: { type: string }[];
    cities: { name: string; id: string }[];
    company: { name: string; slug: string };
  };
}) {
  let navigate = useNavigate();
  const { slug, company, title, cities, countries, remotes } = item;
  return (
    <div>
      <div
        className="job-box"
        onClick={() => navigate(`/about/${slug}/${company?.slug}`)}
      >
        <h3 className="title">{title}</h3>
        <h4 className="company-name">{company?.name}</h4>
        {cities?.map((city, index: number) => (
          <span key={index}>{city.name}</span>
        ))}
        {countries?.map((country, index: number) => (
          <span key={index}> , {country.name}</span>
        ))}

        <p>{remotes?.length > 0 ? "remote" : "onsite"}</p>
      </div>
    </div>
  );
}

export default JobCard;
