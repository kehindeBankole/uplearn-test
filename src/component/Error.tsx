import React from 'react';
import './Loader.scss'
function Loader() {
  return <div>
      <div className="loader-box" data-testid="loader-box"><p>error , failed to fetch</p></div>;
  </div>;
}

export default Loader;
