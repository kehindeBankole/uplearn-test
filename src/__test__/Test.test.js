import React from "react";
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import Loader from "../component/Loader";
import Error from "../component/Error";
import TestRenderer from "react-test-renderer";
import JobCard from "../component/JobCard";
import { BrowserRouter } from "react-router-dom";

test("show loading component", () => {
  const loader = render(<Loader data-testid="loader-box" />);
  const loaderElement = loader.getByTestId("loader-box");
  expect(loaderElement).toBeInTheDocument();
  expect(loaderElement).toHaveClass("loader-box");
  expect(loaderElement).toHaveTextContent("Loading..");
});

test("show error component", () => {
  const error = render(<Error />);
  const errorElement = error.getByTestId("loader-box");
  expect(errorElement).toBeInTheDocument();
  expect(errorElement).toHaveClass("loader-box");
  expect(errorElement).toHaveTextContent("error , failed to fetch");
});

test("take snapshot", () => {
  const item = {
    __typename: "Job",
    title: "Senior Fullstack Engineer - Platform",
    id: "cjz1ipl9x009a0758hg68h7vy",
    company: {
      __typename: "Company",
      name: "Segment",
    },
    cities: [
      {
        __typename: "City",
        name: "San Francisco",
      },
    ],
    countries: [],
    remotes: [],
  };
  const { container } = render(
    <BrowserRouter>
      <JobCard item={item} />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
