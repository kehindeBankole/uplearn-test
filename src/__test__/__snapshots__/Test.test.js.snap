// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take snapshot 1`] = `
<div>
  <div>
    <div
      class="job-box"
    >
      <h3
        class="title"
      >
        Senior Fullstack Engineer - Platform
      </h3>
      <h4
        class="company-name"
      >
        Segment
      </h4>
      <span>
        San Francisco
      </span>
      <p>
        onsite
      </p>
    </div>
  </div>
</div>
`;
